src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/fwd.hpp

```diff
@@ -47,12 +47,372 @@
+///
+#define BSONCXX_POLY_USE_MNMLSTC
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// Defined when configured to use an external mnmlstc/core as the C++17 polyfill library.
+///
+/// @deprecated Support for external polyfill libraries is deprecated and will be removed in an
+/// upcoming major release. Set `ENABLE_BSONCXX_POLY_USE_IMPLS=ON` or `BSONCXX_POLY_USE_IMPLS=ON`
+/// when configuring the bsoncxx library.
+///
+#define BSONCXX_POLY_USE_SYSTEM_MNMLSTC
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// Defined when configured to use Boost as the C++17 polyfill library.
+///
+/// @deprecated Support for external polyfill libraries is deprecated and will be removed in an
+/// upcoming major release. Set `ENABLE_BSONCXX_POLY_USE_IMPLS=ON` or `BSONCXX_POLY_USE_IMPLS=ON`
+/// when configuring the bsoncxx library.
+///
+#define BSONCXX_POLY_USE_BOOST
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// Defined when configured to use bsoncxx's own C++17 polyfill implementations.
+///
+#define BSONCXX_POLY_USE_IMPLS
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// Defined when configured to use the C++ standard library for C++17 features.
+///
+#define BSONCXX_POLY_USE_STD
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// For internal use only!
+///
+/// @deprecated To be removed in an upcoming major release.
+/// <!-- Explicitly documented due to long-standing availability. -->
+///
+#define BSONCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-config-hpp
+/// For internal use only!
+///
+/// @deprecated To be removed in an upcoming major release.
+/// <!-- Explicitly documented due to long-standing availability. -->
+///
+#define BSONCXX_INLINE_NAMESPACE_END
+
+///
+/// @} <!-- bsoncxx/v_noabi/bsoncxx/config/config.hpp -->
+///
+
+///
+/// @defgroup bsoncxx-v_noabi-bsoncxx-config-export-hpp bsoncxx/v_noabi/bsoncxx/config/export.hpp
+/// This page documents macros provided by the generated header
+/// `bsoncxx/v_noabi/bsoncxx/config/export.hpp`.
+///
+/// @name bsoncxx/v_noabi/bsoncxx/config/export.hpp
+///
+/// These macros are provided by @ref bsoncxx-v_noabi-bsoncxx-config-export-hpp "export.hpp",
+/// **NOT** by @ref bsoncxx/v_noabi/bsoncxx/fwd.hpp "fwd.hpp" (where this is documented in source)!
+///
+/// @{
+///
+
+///
+/// @ingroup bsoncxx-v_noabi-bsoncxx-config-export-hpp
+/// Defined when the bsoncxx library is built as a static library.
+///
+/// @warning For internal use only!
+///
+/// This macro is not defined when the bsoncxx library is built as a shared library.
+///
+/// @note This macro is intended to be automatically defined by the CMake or pkg-config package
+/// configuration file used to import the bsoncxx library. Manually defining this macro is NOT
+/// recommended, but may be necessary depending on how the mongocxx library is being imported or
+/// packaged.
+///
+#define BSONCXX_STATIC
```

Comment:
Remove `BSONCXX_STATIC`? The rendered documentation notes "These macros are provided by export.hpp". But `BSONCXX_STATIC` is applied [on the CMake target](https://github.com/mongodb/mongo-cxx-driver/blob/ae07bd2574ae7340f7a047f4dd99186c644a262b/cmake/BsoncxxUtil.cmake#L140), rather than defined in `export.hpp`.

Similar question applies to `MONGOCXX_STATIC`.

Reply:
Good point. Removed as suggested.